src/main/java/uk/gov/digital/ho/hocs/casework/api/BankHolidayService.java

```diff
@@ -0,0 +1,104 @@
+package uk.gov.digital.ho.hocs.casework.api;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.casework.api.dto.BankHolidaysByRegionDto;
+import uk.gov.digital.ho.hocs.casework.application.RestHelper;
+import uk.gov.digital.ho.hocs.casework.client.bankHolidayClient.BankHolidayClient;
+import uk.gov.digital.ho.hocs.casework.client.infoclient.InfoClient;
+import uk.gov.digital.ho.hocs.casework.domain.model.BankHoliday;
+import uk.gov.digital.ho.hocs.casework.domain.repository.BankHolidayRepository;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static net.logstash.logback.argument.StructuredArguments.value;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.*;
+
+@Service
+@Slf4j
+public class BankHolidayService {
+    private final BankHolidayRepository bankHolidayRepository;
+    private final BankHolidayClient bankHolidayClient;
+
+    private final InfoClient infoClient;
+
+    @Autowired
+    public BankHolidayService(final BankHolidayRepository bankHolidayRepository,
+                              final BankHolidayClient bankHolidayClient,
+                              final InfoClient infoClient) {
+        this.bankHolidayRepository = bankHolidayRepository;
+        this.bankHolidayClient = bankHolidayClient;
+        this.infoClient = infoClient;
+    }
+
+    public int refreshBankHolidayTable() {
+        log.info("Refreshing bank holidays");
+
+        final Map<String, BankHolidaysByRegionDto> bankHolidaysByRegion =
+                bankHolidayClient.getBankHolidays();
+        final Set<BankHoliday> currentBankHolidays = bankHolidayRepository.findAll();
+
+        final List<BankHoliday> newBankHolidays = filterOutExistingBankHolidays(bankHolidaysByRegion, currentBankHolidays);
+
+        bankHolidayRepository.saveAll(newBankHolidays);
+        log.info("Saved {} new bank holidays", newBankHolidays.size());
```

Comment:
As we log it out here, I don't see that we need to return a number string to the callee?